packages/styled-components/src/models/StyledComponent.js

```diff
@@ -277,6 +278,8 @@ export default function createStyledComponent(
   });
 
   if (process.env.NODE_ENV !== 'production') {
+    checkDynamicCreation(displayName, styledComponentId);
```

Comment:
Need to add this to keyframes as well yeah?

Reply:
Key frames doesn’t inject its rules on its own anymore and generates a stable name and ID so that ones safe to inline